src/main/java/io/xstefank/wildfly/bot/ConfigFileChangeProcessor.java

```diff
@@ -0,0 +1,103 @@
+package io.xstefank.wildfly.bot;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.difflib.UnifiedDiffUtils;
+import com.github.difflib.patch.Patch;
+import com.github.difflib.patch.PatchFailedException;
+import io.quarkiverse.githubapp.ConfigFile;
+import io.quarkiverse.githubapp.event.PullRequest;
+import io.quarkiverse.githubapp.runtime.UtilsProducer;
+import io.quarkiverse.githubapp.runtime.github.GitHubConfigFileProviderImpl;
+import io.quarkus.logging.Log;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Default;
+import jakarta.inject.Inject;
+import org.kohsuke.github.GHCommitState;
+import org.kohsuke.github.GHEventPayload;
+import org.kohsuke.github.GHPullRequest;
+import org.kohsuke.github.GHPullRequestFileDetail;
+
+import io.xstefank.wildfly.bot.config.RuntimeConstants;
+import io.xstefank.wildfly.bot.config.WildFlyConfigFile;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+@Default
+public class ConfigFileChangeProcessor {
+
+    private static final Pattern DIFF_REGEX = Pattern.compile("^@@\\s+-(?:(\\d+)(?:,(\\d+))?)\\s+\\+(?:(\\d+)(?:,(\\d+))?)\\s+@@", Pattern.MULTILINE);
+
+    @Inject
+    GitHubConfigFileProviderImpl fileProvider;
+
+    @Inject
+    @UtilsProducer.Yaml
+    ObjectMapper yamlObjectMapper;
+
+    void onFileChanged(@PullRequest.Opened @PullRequest.Edited GHEventPayload.PullRequest pullRequestPayload) {
+        GHPullRequest pullRequest = pullRequestPayload.getPullRequest();
+        for (GHPullRequestFileDetail changedFile : pullRequest.listFiles()) {
+            if (changedFile.getFilename().equals(fileProvider.getFilePath(RuntimeConstants.CONFIG_FILE_NAME))) {
+                String stringPatch = adjustDiffFormat(changedFile.getPatch());
+                List<String> lines = Arrays.asList(stringPatch.split("\n"));
+                Patch<String> patch = UnifiedDiffUtils.parseUnifiedDiff(lines);
+
+                String currentFile = fileProvider.fetchConfigFile(pullRequestPayload.getRepository(), RuntimeConstants.CONFIG_FILE_NAME, ConfigFile.Source.DEFAULT, String.class).get();
+                List<String> currentFileLines = Arrays.asList(currentFile.split("\n"));
+                try {
+                    String updatedFile = patch.applyTo(currentFileLines)
+                            .stream()
+                            .collect(Collectors.joining("\n"));
+                    Optional<WildFlyConfigFile> file = Optional.ofNullable(yamlObjectMapper.readValue(updatedFile, WildFlyConfigFile.class));
+                    List<String> invalidRules = validateFile(file.get());
```

Comment:
If the file already parses it should be validated. Why do we not marking ids as required?

Reply:
Now we have a few fields, which we can validate using `hibernate-validator`. id, email or description. I think we should create a separate issue for that.